workshops/sprig_first_game/README.md

```diff
@@ -0,0 +1,479 @@
+---
+name: "Build your first Sprig game!"
+description: "Get started with Sprig and build a game!"
+author: "@jianmin-chen"
+img: "https://user-images.githubusercontent.com/27078897/186008909-cc9ea9d5-5843-487e-ac3a-29330496eed1.png"
+---
+
+Let's make our first game in Sprig! 
+
+[Sprig](https://editor.sprig.hackclub.com/) is a tool developed by Hack Club to help people quickly get started making interesting tile games. It's a tiny game engine embedded in a web-based editor with integrated art and music tools. If you build a game in Sprig and share it in the community gallery you can get a [Sprig Console](https://sprig.hackclub.com)!
+
+We're going to build a Sokoban game to get started with Sprig. Let's open up Sprig first:
+
+![Sprig's editor](https://cloud-mju5h5o89-hack-club-bot.vercel.app/0screenshot_2022-11-02_at_16-01-13_sprig.png)
+
+On the left, we have an editor, and on the right, we have the actual game. We're going to be using the editor to write the code for our game.
+
+## Creating a Player
+
+To start, let's assign a single character to a variable that we can use in our spite legend.
+
+```js
+const player = "p";
+```
+
+We can then set the image we want to use for our player, like so:
+
+```js
+const player = "p";
+
+setLegend(
+    [ player, bitmap`...` ]
+)
+```
+
+`setLegend` allows us to set up a sprite for our character. We can click on `bitmap` to open up the pixel editor and draw an image for our sprite.
+
+![Bitmap editor](https://user-images.githubusercontent.com/27078897/197604643-2a59cc85-5a07-446d-95b3-d9be844b62c0.gif)
+
+The bitmap is stored as a string. To take a look at it, click on the little arrow next to the lint number. You can minimize it by clicking the area again.
+
+Awesome! Now let's add some obstacles, boxes, and objectives to our game:
+
+```js
+const player = "p";
+const box = "b";
+const goal = "g";
+const wall = "w";
+
+setLegend(
+  [ player, bitmap`
+................
+................
+................
+.......0........
+.....00.000.....
+....0.....00....
+....0.0.0..0....
+....0......0....
+....0......0....
+....00....0.....
+......00000.....
+......0...0.....
```

Comment:
Could we give this section a bit more context? The code block here is massive :D

Reply:
No problem! I'll write up an explanation for it.